exonum-java-binding/core/rust/src/cmd/mod.rs

```diff
@@ -0,0 +1,627 @@
+    /// Run the node in development mode (generate configuration and db files automatically).
+    ///
+    /// Runs one node with public API address 127.0.0.1:8080, private API address 127.0.0.1:8081,
+    /// EJB port 6400.
+    #[structopt(name = "run-dev")]
+    RunDev(RunDev),
+    /// Perform different maintenance actions.
+    #[structopt(name = "maintenance")]
+    Maintenance(Maintenance),
+}
+
+impl Command {
+    /// Parse arguments from the command line arguments.  Print the
+    /// error message and quit the program in case of failure.
+    pub fn from_args() -> Self {
+        <Self as StructOpt>::from_args()
+    }
+}
+
+impl EjbCommand for Command {
+    fn execute(self) -> Result<EjbCommandResult, failure::Error> {
+        match self {
+            Command::GenerateTemplate(c) => c.execute().map(Into::into),
+            Command::GenerateConfig(c) => c.execute().map(Into::into),
+            Command::Finalize(c) => c.execute().map(Into::into),
+            Command::Run(c) => c.execute(),
+            Command::RunDev(c) => c.execute(),
+            Command::Maintenance(c) => c.execute().map(Into::into),
+        }
+    }
+}
+
+/// EJB-specific `generate-template` command. Takes an additional
+/// `supervisor_mode` parameter and saves it in the generated config file.
+#[derive(Debug, StructOpt, Serialize, Deserialize)]
+#[structopt(rename_all = "kebab-case")]
+pub struct GenerateTemplate {
+    #[structopt(flatten)]
+    #[serde(flatten)]
+    standard: StandardGenerateTemplate,
+    /// Supervisor service mode. Can be "simple" or "decentralized".
+    #[structopt(long)]
+    supervisor_mode: SupervisorMode,
+}
+
+/// EJB-specific `generate-config` command. Only does passing `supervisor_mode`
+/// from common config to node public config.
+#[derive(Debug, StructOpt, Serialize, Deserialize)]
+pub struct GenerateConfig {
+    #[structopt(flatten)]
+    standard: StandardGenerateConfig,
+}
+
+/// EJB-specific `finalize` command. Only does passing `supervisor_mode` from
+/// node public configs to final node configuration. Returns error in case
+/// supervisor mode isn't equal in nodes public configs.
+#[derive(Debug, StructOpt, Serialize, Deserialize)]
+pub struct Finalize {
+    #[structopt(flatten)]
+    standard: StandardFinalize,
+}
+
+impl Finalize {
+    /// Returns `SupervisorMode` from public configs. Returns error if
+    /// `SupervisorMode` is not equal in every public config.
+    fn supervisor_mode(&self) -> Result<SupervisorMode, failure::Error> {
+        let mut public_configs: Vec<EjbSharedConfig> = Vec::new();
+        for path in &self.standard.public_configs {
+            public_configs.push(load_config_file(path)?);
+        }
+        let supervisor_mode = public_configs[0].supervisor_mode;
+
+        if public_configs
+            .iter()
+            .any(|config| config.supervisor_mode != supervisor_mode)
+        {
+            return Err(format_err!("Different supervisor modes in public configs"));
```

Comment:
I'd include expected/actual or just a set of the modes, as we can't tell what's expected.